gtsam/base/serializationTestHelpers.h

```diff
@@ -50,11 +58,22 @@ void roundtrip(const T& input, T& output) {
   deserialize(serialized, output);
 }
 
+// Templated round-trip serialization using a file
+template<class T>
+void roundtripFile(const T& input, T& output) {
+  std::string path = resetFilesystem();
+
+  serializeToFile(input, path + "graph.dat");
+  deserializeFromFile(path + "graph.dat", output);
```

Comment:
Tests are failing over here because of some issue with Boost's deserialization. In Boost 1.58, deserializing a vector of vectors causes the data to be flattened into a single vector, hence the equality check fails (e.g. the test checks for a 3-vector of 2-vectors vs a 6-vector) . This does not happen in newer versions of Boost (e.g. 1.59) so it definitely seems to be a bug in Boost that needs to be worked around. I am not sure what the root cause of the bug is, so I can't recommend a long term fix. Stackoverflow/Google isn't helpful either.

Alternatively, we could not have these file based checks (save it for a later PR when we bump the boost version) and let the user deal with the consequences of their choice of Boost version.

Reply:
Fixed this actually
Found the bug which caused this - I was reusing the `output` object between tests so I guess there was something weird with the size being preassigned or something.  By just creating a new object for each test, the vectors are created the right size.  My mistake here.